kedro/ipython/__init__.py

```diff
@@ -178,3 +185,134 @@ def _find_kedro_project(current_dir: Path) -> Any:  # pragma: no cover
         current_dir = current_dir.parent
 
     return None
+
+
+@typing.no_type_check
+@magic_arguments()
+@argument(
+    "node",
+    type=str,
+    help=("Name of the Node."),
+    nargs="?",
+    default=None,
+)
+def magic_load_node(node: str) -> None:
+    """The line magic %load_node <node_name>
+    Currently it only supports Jupyter Notebook (>7.0) and Jupyter Lab. This line magic
+    will generate code in multiple cells to load datasets from `DataCatalog`, import
+    relevant functions and modules, node function definition and a function call.
+    """
+    cells = _load_node(node, pipelines)
+    from ipylab import JupyterFrontEnd
+
+    app = JupyterFrontEnd()
+
+    def _create_cell_with_text(text: str) -> None:
+        # Noted this only works with Notebook >7.0 or Jupyter Lab. It doesn't work with
+        # VS Code Notebook due to imcompatible backends.
+        app.commands.execute("notebook:insert-cell-below")
+        app.commands.execute("notebook:replace-selection", {"text": text})
+
+    for cell in cells:
+        _create_cell_with_text(cell)
+
+
+def _load_node(node_name: str, pipelines: _ProjectPipelines) -> list[str]:
+    """Prepare the code to load dataset from catalog, import statements and function body.
+
+    Args:
+        node_name (str): The name of the node.
+
+    Returns:
+        list[str]: A list of string which is the generated code, each string represent a
+        notebook cell.
+    """
+    warnings.warn(
+        "This is an experimental feature, only Jupyter Notebook (>7.0) & Jupyter Lab "
+        "are supported. If you encounter unexpected behaviour or would like to suggest "
+        "feature enhancements, add it under this github issue https://github.com/kedro-org/kedro/issues/3580"
+    )
+    node = _find_node(node_name, pipelines)
+    node_func = node.func
+
+    node_inputs = _prepare_node_inputs(node)
+    imports = _prepare_imports(node_func)
+    function_definition = _prepare_function_body(node_func)
+    function_call = _prepare_function_call(node_func)
+
+    cells: list[str] = []
+    cells.append(node_inputs)
+    cells.append(imports)
+    cells.append(function_definition)
+    cells.append(function_call)
+    return cells
+
+
+def _find_node(node_name: str, pipelines: _ProjectPipelines) -> Node:
+    for pipeline in pipelines.values():
+        try:
+            found_node: Node = pipeline.filter(node_names=[node_name]).nodes[0]
+            return found_node
+        except ValueError:
+            continue
+    # If reached the node was not found in the project
+    raise ValueError(
```

Comment:
I wonder if we want to support `func_name` at some point too? 🤔

Reply:
This was briefly discussed in last TD. @AhdraMeraliQB originally have an implementation. 

It runs into issue if function is used twice, nodes are unique but functions do not have to. Fundamentally we want to fix this with the default node name but this was decided to be fixed later.